modules/host/network_test.go

```diff
@@ -66,37 +66,43 @@ func TestHostWorkingStatus(t *testing.T) {
 		t.SkipNow()
 	}
 	t.Parallel()
+
 	ht, err := newHostTester(t.Name())
 	if err != nil {
 		t.Fatal(err)
 	}
+	defer ht.Close()
 
 	if ht.host.WorkingStatus() != modules.HostWorkingStatusChecking {
 		t.Fatal("expected working state to initially be modules.HostWorkingStatusChecking")
 	}
 
-	// Simulate some setting calls, and see if the host picks up on it.
-	atomic.AddUint64(&ht.host.atomicSettingsCalls, workingStatusThreshold+1)
-	time.Sleep(workingStatusFirstCheck + time.Second)
-	if ht.host.WorkingStatus() != modules.HostWorkingStatusWorking {
-		t.Fatal("expected host working status to be modules.HostWorkingStatusWorking after incrementing status calls")
-	}
+	for i := 0; i < 5; i++ {
+		// Simulate some setting calls, and see if the host picks up on it.
+		atomic.AddUint64(&ht.host.atomicSettingsCalls, workingStatusThreshold+1)
 
-	// No more settings calls, host should believe it is not working now.
-	time.Sleep(workingStatusFrequency + time.Second)
-	if ht.host.WorkingStatus() != modules.HostWorkingStatusNotWorking {
-		t.Fatal("expected host working status to be modules.HostWorkingStatusNotWorking after waiting workingStatusFrequency with no settings calls")
-	}
+		success := false
+		for start := time.Now(); time.Since(start) < 30*time.Second; time.Sleep(time.Millisecond * 10) {
```

Comment:
ah, `build.Retry` was added in the #1758. I can push that commit to master if you want. Even if we merge #1758, `build.Retry` won't make it into master until we merge the gateway-proto branch.